presto-main/src/main/java/com/facebook/presto/metadata/BuiltInTypeAndFunctionNamespaceManager.java

```diff
@@ -682,6 +687,7 @@ private List<? extends SqlFunction> getBuildInFunctions(FeaturesConfig featuresC
                 .aggregates(ClassificationPrecisionAggregation.class)
                 .aggregates(ClassificationRecallAggregation.class)
                 .aggregates(ClassificationThresholdsAggregation.class)
+                .functions(ARRAY_CUM_SUM_BIG_INT, ARRAY_CUM_SUM_DOUBLE, ARRAY_CUM_SUM_INTEGER, ARRAY_CUM_SUM_REAL, ARRAY_CUM_SUM_GENERIC)
```

Comment:
Have to create five functions, this is because this function is special that the output type is not the same as the input type (hence cannot use typeParameters and parametric scalar functions), and the return type depends on the type of input (hence cannot simply fixing the return type).

We have similar functions, for example the sum aggregations, where when input is integer/bigint, output is bigint. However, this is achieved by only implementing the sum aggregation for bigint and not for integer. The input will be cast to bigint in sql level. But this is not what we want, as cast array(integer) to array(bigint) is expensive.

I tried several ways, and this is the best way to work around these constraints.